templates/web.letsencrypt.ssl.template.yml

```diff
@@ -0,0 +1,58 @@
+env:
+  LETSENCRYPT_DIR: "/shared/letsencrypt"
+
+hooks:
+  after_ssl:
+    - exec:
+       cmd:
+         - if [ -z "$LETSENCRYPT_ACCOUNT_EMAIL" ]; then echo "LETSENCRYPT_ACCOUNT_EMAIL ENV variable is required and has not been set."; exit 1; fi
+         - /bin/bash -c "if [[ ! \"$LETSENCRYPT_ACCOUNT_EMAIL\" =~ ([^@]+)@([^\.]+) ]]; then echo \"LETSENCRYPT_ACCOUNT_EMAIL is not a valid email address\"; exit 1; fi"
+
+    - exec:
+       cmd:
+         - apt-get install -y netcat
+         - cd /root && git clone https://github.com/Neilpang/acme.sh.git && cd /root/acme.sh && git reset --hard 8d5618c44a2ab973aa7eb243db740e22c742b809
```

Comment:
ACME is the tool we use for requesting the certs and getting them approved. See: https://github.com/Neilpang/acme.sh for more information.

Pulling it down from an "external" resource does not bother me that much to be honest, since we lock it down to a "version" (the git ref). We pull almost everything down from external resources, even Dokku. This library is also used by way bigger tools like Discourse.

I feel confident by pulling this over git. another option would be to download this https://get.acme.sh/ file and execute that, but that would open up the exact same problems.